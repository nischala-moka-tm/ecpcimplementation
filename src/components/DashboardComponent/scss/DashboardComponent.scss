@import "../../../shared/scss/Common-Style.scss";

.recent-activity,
.recent-updates {
  overflow-x: auto;

  @media (max-width: $breakpoint-tablet) {
    margin-top: 30px;
  }

  p {
    margin-left: 15px;
    font-family: $toyota-bold;
  }

  table {
    width: 100%;
    font-family: $toyota-regular;

    th,
    td {
      padding: 10px 5px;

      @media (min-width: $breakpoint-tablet) {
        padding: 10px;
      }
    }

    th {
      background: #D4D2D2 0% 0% no-repeat padding-box;
      font-size: 15px;
      line-height: 16px;
      white-space: nowrap;

      @media (min-width: $breakpoint-tablet) {
        font-size: 15px;
        line-height: 22px;
      }
    }

    td {
      font-size: 10px;
      line-height: 12px;

      @media (min-width: $breakpoint-tablet) {
        font-size: 13px;
        line-height: 20px;
      }

      &.edit,
      &.view {
        cursor: pointer;
        color: #2a79fd;
      }
    }

    tr {
      &:nth-child(even) {
        background: #E3E3E3 0% 0% no-repeat padding-box;
      }
    }
  }
}


.recent-updates {
  margin-top: 40px;
}

.recent-activity-list {
  text-align: right;
  padding: 0 1.5rem 0 0;
  font-size: 14px;

  a {
    margin-left: 5px;
  }
}

.recent-activity-list-one {
  text-align: right;
  padding: 0 1.5rem 0 0;
  margin-bottom: 1rem;
  font-size: 14px;

  a {
    margin-left: 5px;
  }
}