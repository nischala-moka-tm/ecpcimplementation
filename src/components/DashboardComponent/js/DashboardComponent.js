import React, { useState, useEffect } from "react";
import "../scss/DashboardComponent.scss";
import { FaRegEdit, FaRegEye } from "react-icons/fa";
import { dateFormat } from "../../CommonBlocks/js/CommonBlock";
import { AxiosGet, AxiosPostMetadata } from "../../AxiosMethods/ApiCalls";
import AddPermissionLevels from "../../AddPermissionLevels/js/AddPermissionLevels";

import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

const userRole = "ECPC_TOYOTA_ADMIN";
const userId = 1234567;
const navalue = "N/A";

const checkUndefined = (value) => {
  return value ? value : navalue;
};

const checkDateUndefined = (datevalue) => {
  return datevalue ? dateFormat(datevalue) : navalue;
};
function DashboardComponent(props) {
  const [recentUpdateData, setRecentUpdateData] = useState([]);
  const [isNoDataFound, setNoDataFound] = useState(false);
  const [showViewLevel, SetShowViewLevel] = useState(false);
  const [showModifyLevel, SetShowModifyLevel] = useState(false);

  useEffect(() => {
    getApiCall();
  }, [props.brand, props.type]);

  const getApiCall = () => {
    const getDataApi = AxiosGet({
      brand: props.brand,
      type: `${props.type}&userId=${userId}&role=${userRole}`,
    });
    getDataApi.then((result) => {
      result.data.data ? LoadData(result.data.data) : setNoDataFound(true);
    });
  };
  const LoadData = (data) => {
    setRecentUpdateData(data.recentUpdate);
  };

  const [resposnsedata, setData] = useState([recentUpdateData]);
  const modifyData = async (data, brand) => {
    const finalData = {
      adminMetaData: {
        id: data.id,
        brand: brand,
      },
    };
    const getMetaData = await AxiosPostMetadata(finalData);
    setData(getMetaData.data);
  };
  const notify = (res, resType) => {
    if (resType === "success") {
      toast.success(res);
      getApiCall();
    } else {
      toast.error(res);
    }
  };
  return (
    <div className="dashboard-component" id="dashboard-page">
      <ToastContainer />
      <div className="recent-activity">
        <p>My Recent Activities</p>
        <table className="table">
          <thead>
            <tr>
              <th>Category</th>
              <th>Level 2</th>
              <th>Level 3</th>
              <th>Level 4</th>
              <th>Level 5</th>
              <th>Created Date</th>
              <th>Last Modified</th>
              <th>Status</th>
              <th>Modify</th>
            </tr>
          </thead>
          <tbody>
            {recentUpdateData &&
              recentUpdateData.slice(0, 4).map((recentlist, index) => {
                return (
                  <tr key={index}>
                    <td>{checkUndefined(recentlist.levels[0])}</td>
                    <td>{checkUndefined(recentlist.levels[1])}</td>
                    <td>{checkUndefined(recentlist.levels[2])}</td>
                    <td>{checkUndefined(recentlist.levels[3])}</td>
                    <td>{checkUndefined(recentlist.levels[4])}</td>
                    <td>{checkDateUndefined(recentlist.createdDate)}</td>
                    <td>{checkDateUndefined(recentlist.modifiedDate)}</td>
                    <td>{checkUndefined(recentlist.status)}</td>
                    <td
                      className="edit"
                      onClick={(e) => {
                        SetShowModifyLevel(true);
                        modifyData(recentlist, props.brand);
                      }}
                    >
                      <FaRegEdit />
                    </td>
                  </tr>
                );
              })}
          </tbody>
        </table>
        {isNoDataFound && <strong>No Data Found</strong>}
        {showModifyLevel && resposnsedata && resposnsedata.level && (
          <AddPermissionLevels
            show={showModifyLevel}
            onClose={() => {
              SetShowModifyLevel(false);
              setData([]);
            }}
            category={resposnsedata}
            optionType="Edit"
            brand={props.brand}
            level={resposnsedata.level}
            notify={notify}
          />
        )}
      </div>
      <div className="recent-activity-list">
        To See all My activities
        <a href="/dashboard-admin/report-audit">Click here</a>
      </div>
      <div className="recent-updates">
        <p>Recent Admin Portal Activities</p>
        <table className="table">
          <thead>
            <tr>
              <th>Category</th>
              <th>Level 2</th>
              <th>Level 3</th>
              <th>Level 4</th>
              <th>Level 5</th>
              <th>Created Date</th>
              <th>Last Modified By</th>
              <th>Last Modified</th>
              <th>Status</th>
              <th>View</th>
            </tr>
          </thead>
          <tbody>
            {recentUpdateData &&
              recentUpdateData.slice(0, 4).map((activitylist, index) => {
                return (
                  <tr key={index}>
                    <td>{checkUndefined(activitylist.levels[0])}</td>
                    <td>{checkUndefined(activitylist.levels[1])}</td>
                    <td>{checkUndefined(activitylist.levels[2])}</td>
                    <td>{checkUndefined(activitylist.levels[3])}</td>
                    <td>{checkUndefined(activitylist.levels[4])}</td>
                    <td>{checkDateUndefined(activitylist.createdDate)}</td>
                    <td>{checkUndefined(activitylist.createdBy)}</td>
                    <td>{checkDateUndefined(activitylist.modifiedDate)}</td>
                    <td>{checkUndefined(activitylist.status)}</td>
                    <td
                      className="view"
                      onClick={() => {
                        SetShowViewLevel(true);
                        modifyData(activitylist, props.brand);
                      }}
                    >
                      <FaRegEye />
                    </td>
                  </tr>
                );
              })}
          </tbody>
        </table>
        {isNoDataFound && <strong>No Data Found</strong>}
        {showViewLevel && resposnsedata && resposnsedata.level && (
          <AddPermissionLevels
            show={showViewLevel}
            onClose={() => {
              SetShowViewLevel(false);
              setData([]);
            }}
            category={resposnsedata}
            optionType="View"
            brand={props.brand}
            level={resposnsedata.level}
            notify={notify}
          />
        )}
      </div>
      <div className="recent-activity-list-one">
        To See all Admin portal activities
        <a href="/dashboard-admin/report-perm">Click here</a>
      </div>
    </div>
  );
}

export default DashboardComponent;
