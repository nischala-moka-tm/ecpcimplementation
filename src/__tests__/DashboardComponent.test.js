import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
Enzyme.configure({ adapter: new Adapter() });
import { act } from "react-dom/test-utils";

import DashboardComponent from "../components/DashboardComponent/js/DashboardComponent";

it("renders correctly when there are no items", () => {
  const tree = renderer
    .create(<DashboardComponent brand="Toyota" type="dashboard" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("renders <DashboardComponent />", () => {
  it("should pass mount snapshot test", async () => {
    let wrapper;
    await act(async () => {
      wrapper = mount(<DashboardComponent brand="Toyota" type="dashboard" />);
    });
    expect(wrapper).toMatchSnapshot();
  });
});
