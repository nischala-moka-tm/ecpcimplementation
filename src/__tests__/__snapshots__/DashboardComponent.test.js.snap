// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <DashboardComponent /> should pass mount snapshot test 1`] = `ReactWrapper {}`;

exports[`renders correctly when there are no items 1`] = `
<div
  className="dashboard-component"
  id="dashboard-page"
>
  <div
    className="Toastify"
  />
  <div
    className="recent-activity"
  >
    <p>
      My Recent Activities
    </p>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Category
          </th>
          <th>
            Level 2
          </th>
          <th>
            Level 3
          </th>
          <th>
            Level 4
          </th>
          <th>
            Level 5
          </th>
          <th>
            Created Date
          </th>
          <th>
            Last Modified
          </th>
          <th>
            Status
          </th>
          <th>
            Modify
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="recent-activity-list"
  >
    To See all My activities
    <a
      href="/dashboard-admin/report-audit"
    >
      Click here
    </a>
  </div>
  <div
    className="recent-updates"
  >
    <p>
      Recent Admin Portal Activities
    </p>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Category
          </th>
          <th>
            Level 2
          </th>
          <th>
            Level 3
          </th>
          <th>
            Level 4
          </th>
          <th>
            Level 5
          </th>
          <th>
            Created Date
          </th>
          <th>
            Last Modified By
          </th>
          <th>
            Last Modified
          </th>
          <th>
            Status
          </th>
          <th>
            View
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="recent-activity-list-one"
  >
    To See all Admin portal activities
    <a
      href="/dashboard-admin/report-perm"
    >
      Click here
    </a>
  </div>
</div>
`;
